### ISSUE TITLE
Complete rewrite?

### ISSUE THREAD

**Issue by Firedrake969** [ScratchAgarioBots/agario-bot#10]
I'm thinking we might just want to write our own from scratch.  There's no documentation on this that I can find, formatting is nonexistent, and we'll understand the bot better if we make it ourselves.  Thoughts?

**Comment by Thepuzzlegame** [comment]
I'm in favor of this.

**Comment by Firedrake969** [comment]
thanks, this is driving me crazy :P  but I must sleep

note to self - stop propagation/prevent default of main script once loaded then load own modified script
see greasyfork top agar bot for example

**Comment by PullJosh** [comment]
I say just AI.js. It _really_ helps to have all the data already given to you, and we can take that and put it into whatever form we'd like (for example, we could put all the data into a single object, or we could use all separate variables, or sort it out into a few objects... etc.) But we wouldn't have to go dig through the (seemingly non-existant) agar.io source.

**Comment by an-OK-squirrel** [comment]
If you rewrite it, you could make it a userscript. That would make it cross-platform.

**Comment by PullJosh** [comment]
The more that I think about this, the better a rewrite is sounding.

**Comment by Firedrake969** [comment]
Yeah, I was just thinking about rewriting Ai.js.

**Comment by an-OK-squirrel** [comment]
Would it be possible to add a userscript version of script.js that works the same?

**Comment by Firedrake969** [comment]
Probably, but we should first figure out how to get the data :P  I'd say that's the most important part

**Comment by an-OK-squirrel** [comment]
Yep. I'm going to go look and then not figure it out and then come back and tell you that I didn't figure out anything. :P

**Comment by Firedrake969** [comment]
I tried that last night xD

**Comment by an-OK-squirrel** [comment]
I didn't find anything!

**Comment by an-OK-squirrel** [comment]
I can't because I think the stuff only works when you're off the menu and it's hard to play and inspect element at the same time. :P

**Comment by Firedrake969** [comment]
I don't play when I look and I still couldn't :L

**Comment by an-OK-squirrel** [comment]
It's on a canvas. D:

**Comment by Firedrake969** [comment]
https://github.com/Apostolique/Agar.io-bot seems like is has all the functions we want

**Comment by an-OK-squirrel** [comment]
Ok, checking it out

**Comment by an-OK-squirrel** [comment]
Oh, its even a userscript!

**Comment by Firedrake969** [comment]
yeah we just mod it.  I'll fork it

**Comment by an-OK-squirrel** [comment]
But what part grabs the data?

**Comment by Firedrake969** [comment]
getPlayerData() :P

**Comment by an-OK-squirrel** [comment]
Oh :P What does the other stuff do?

**Comment by Firedrake969** [comment]
getSomethingHere()

**Comment by an-OK-squirrel** [comment]
Oh. :P

**Comment by an-OK-squirrel** [comment]
So, bot.js is the thing we need to modify?

**Comment by Firedrake969** [comment]
yeah but launcher.js is the important data grabbing stuff.

**Comment by an-OK-squirrel** [comment]
As I thought.

**Comment by an-OK-squirrel** [comment]
Also, can we delete -rewritten? It's useless now. :P

**Comment by Firedrake969** [comment]
bot.user.js findDestination() is what we want to modify, and we should delete the other two repos

**Comment by PullJosh** [comment]
Wait!

**Comment by PullJosh** [comment]
Keep the first one, the non-rewritten one, just in case this goes badly.

**Comment by Firedrake969** [comment]
Ok

**Comment by an-OK-squirrel** [comment]
Thanks. Moving this conversation to the other repo..

**Comment by PullJosh** [comment]
I haz no permissions on the new one.

Which is weird.

**Comment by Firedrake969** [comment]
I'll add it

**Comment by PullJosh** [comment]
Thanks!

**Comment by an-OK-squirrel** [comment]
I made an issue on the new one
